include/loads_bcs/cohesion_constraint.h

```diff
@@ -0,0 +1,66 @@
+#ifndef MPM_COHESION_CONSTRAINT_H_
+#define MPM_COHESION_CONSTRAINT_H_
+
+#include "data_types.h"
+
+namespace mpm {
+
+//! CohesionConstraint class to store cohesion constraint on a set
+//! \brief CohesionConstraint class to store a constraint on a set
+//! \details CohesionConstraint stores the constraint as a static value
+class CohesionConstraint {
+ public:
+  // Constructor
+  //! \param[in] setid  set id
+  //! \param[in] dir Direction of constraint load (normal)
+  //! \param[in] sign_n Sign of normal vector
+  //! \param[in] cohesion Constraint cohesion
+  //! \param[in] h_min Characteristic length (cell height)
+  //! \param[in] nposition Nodal location, nposition, along boundary
+  CohesionConstraint(int setid, unsigned dir, int sign_n, double cohesion,
+                     double h_min, int nposition = 0)
+      : setid_{setid},
+        dir_{dir},
+        sign_n_{sign_n},
+        cohesion_{cohesion},
+        h_min_{h_min},
+        nposition_{nposition} {};
+
+  // Set id
+  int setid() const { return setid_; }
+
+  // Direction
+  unsigned dir() const { return dir_; }
+
+  // Sign of normal direction
+  int sign_n() const { return sign_n_; }
+
+  // Return cohesion
+  double cohesion() const { return cohesion_; }
+
+  // Cell height
+  double h_min() const { return h_min_; }
+
+  // Return nposition
+  int nposition() const { return nposition_; }
+  // 0: None (no nposition is specified)
+  // 1: Corner (nodes at boundary corners)
+  // 2: Edge (nodes along boundary edges)
+  // 3: Face (nodes on boundary faces)
+
+ private:
+  // ID
+  int setid_;
+  // Direction (normal)
+  unsigned dir_;
+  // Sign of normal direction
+  int sign_n_;
+  // Cohesion
+  double cohesion_;
+  // Cell height
+  double h_min_;
+  // Node nposition
+  int nposition_;
```

Comment:
Above mentions that `nposition_` default is `0`. It is not initialized as zero here. I don't see its assigned anywhere else.

It seems if I set `nposotion_ = 5`, the switch statement below throws an error. So there is no need for a no-position specific value? Am I following this correctly?

Reply:
I think the error will be thrown. We probably can remove the comment "_0: None (no nposition is specified)_" to make it clearer.

Reply:
I think the only scenario where anything would come up is if somebody is reading constraints from a file and forgets to include the last column of positional `int`s. 

```
istream >> id >> dir >> sign_n >> cohesion >> h_min >> nposition;
```

I think the istream extractor will just not re-assign nposition which is initialized without a value above. So maybe there is chance that the nposition value is accidentally 1, 2, or 3.

Reply:
I am not sure if someone forget to add the last column. I guess three things might happen: (i) `std::istream` will throw, (ii) wont read anything, (iii) might read the next line first column (unlikely). If (i) it is good, since the user is warned due to incompatible input. If (ii) it is ok since if `nposition` is not 1, 2, or 3, the error message at `node.tcc` will be thrown. If (iii), then this might be an issue, since the next line first element is `id`, which is also an unsigned/int. Is this answering your concern, @jgiven100? @LDTalbot can you give it a try?

Reply:
Oh, I am not sure the default value will be 1,2,or 3. It will be most likely a random "garbage" value.

Reply:
https://www.learncpp.com/cpp-tutorial/uninitialized-variables-and-undefined-behavior/

Reply:
@bodhinandach Yes. As indicated in [this comment](https://github.com/geomechanics/mpm/pull/65#discussion_r966373301), it seems `istream >>` does what you call option (ii) when the last column is missing. The "garbage" initialization value is what is passed along in the code. I agree it seems very unlikely the "garbage" value will just so happen to be 1, 2, or 3,

@LDTalbot will push 
```
int nposition = 0;
```
to resolve.

Reply:
So it seems that it should be OK...

Reply:
I am not sure we need to check it all the time we read. That's going to cost some time. Other read functions are not having this too. The one implemented in node.tcc should serve as a check already.

Reply:
Theres no additional checks. Just guaranteed to have `nposition=0` for the case that I mess up input files.

Reply:
Sounds good to me. Let's put:
`int nposition = 0;`